pkg/providers/instancetype/types.go

```diff
@@ -184,6 +196,12 @@ func ephemeralStorage(amiFamily amifamily.AMIFamily, blockDeviceMappings []*v1al
 			}
 		}
 	}
+	//Return the ephemeralBlockDevice size if defined in ami
+	if ephemeralBlockDevice, ok := lo.Find(amiFamily.DefaultBlockDeviceMappings(), func(item *v1alpha1.BlockDeviceMapping) bool {
```

Comment:
I'm not sure that this is correct. I think we need to get the last `DefaultBlockDeviceMapping` to consider that one for the ephemeral-storage capacity. In general, we are taking the last blockDeviceMapping to be the ephemeral block device.

Reply:
So, what's the purpose of AMIFamily.EphemeralBlockDevice?
Is it the device name of the ephemeral storage?

Reply:
Yep, you are completely right. I didn't realize that this was a re-write of existing code. Good call